src/main/java/co/cask/hydrator/salesforce/plugin/source/streaming/SalesforcePushTopicListener.java

```diff
@@ -0,0 +1,141 @@
+ * the License.
+ */
+
+package co.cask.hydrator.salesforce.plugin.source.streaming;
+
+import co.cask.hydrator.salesforce.SalesforceConstants;
+import co.cask.hydrator.salesforce.authenticator.AuthResponse;
+import co.cask.hydrator.salesforce.authenticator.Authenticator;
+import co.cask.hydrator.salesforce.authenticator.AuthenticatorCredentials;
+import org.awaitility.Awaitility;
+import org.awaitility.core.ConditionTimeoutException;
+import org.cometd.client.BayeuxClient;
+import org.cometd.client.transport.LongPollingTransport;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Listens to a specific Salesforce pushTopic and adds messages to the blocking queue,
+ * which can be read by a user of the class.
+ */
+public class SalesforcePushTopicListener {
+  private static final Logger LOG = LoggerFactory.getLogger(SalesforcePushTopicListener.class);
+
+  private static final String DEFAULT_PUSH_ENDPOINT = "/cometd/" + SalesforceConstants.API_VERSION;
+  /**
+   * Timeout of 110 seconds is enforced by Salesforce Streaming API and is not configurable.
+   * So we enforce the same on client.
+   */
+  private static final int CONNECTION_TIMEOUT = 110;
+  private static final long HANDSHAKE_TIMEOUT_MS = TimeUnit.SECONDS.toMillis(110);
+
+  private static final int HANDSHAKE_CHECK_INTERVAL_MS = 1000;
+
+  // store message string not JSONObject, since it's not serializable for later Spark usage
+  private final BlockingQueue<String> messagesQueue = new LinkedBlockingQueue<>();
+
+  private final AuthenticatorCredentials credentials;
+  private final String topic;
+
+  public SalesforcePushTopicListener(AuthenticatorCredentials credentials, String topic) {
+    this.credentials = credentials;
+    this.topic = topic;
+  }
+
+  /**
+   * Start the Bayeux Client which listens to the Salesforce PushTopic and saves received messages
+   * to the queue.
+   */
+  public void start() {
+    try {
+      BayeuxClient bayeuxClient = getClient(credentials);
+      waitForHandshake(bayeuxClient, HANDSHAKE_TIMEOUT_MS, HANDSHAKE_CHECK_INTERVAL_MS);
+      LOG.debug("Client handshake done");
+      bayeuxClient.getChannel("/topic/" + topic).subscribe((channel, message) -> {
+        messagesQueue.add(message.getJSON());
+      });
+
+    } catch (Exception e) {
+      throw new RuntimeException("Could not start client", e);
+    }
+  }
+
+  /**
+   * Retrieves message from the messages queue, waiting up to the
+   * specified wait time if necessary for an element to become available.
+   *
+   * @param timeout how long to wait before giving up
```

Comment:
what unit is timeout in? Let's include in variable name

Reply:
The unit is next argument

   * @param timeout how long to wait before giving up
   * @param unit timeunit of timeout

So it can be seconds milliseconds or what ever you set it to.